crates/compiler/can/src/derive.rs

```diff
@@ -213,6 +213,48 @@ fn is_eq<'a>(env: &mut Env<'a>, at_opaque: &'a str) -> ast::Expr<'a> {
     )
 }
 
+fn to_inspector<'a>(env: &mut Env<'a>, at_opaque: &'a str) -> ast::Expr<'a> {
+    let alloc_pat = |it| env.arena.alloc(Loc::at(DERIVED_REGION, it));
+    let alloc_expr = |it| env.arena.alloc(Loc::at(DERIVED_REGION, it));
+
+    let payload = "#payload";
+
+    // \@Opaq payload
+    let opaque_ref = alloc_pat(ast::Pattern::OpaqueRef(at_opaque));
+    let opaque_apply_pattern = ast::Pattern::Apply(
+        opaque_ref,
+        &*env
+            .arena
+            .alloc([Loc::at(DERIVED_REGION, ast::Pattern::Identifier(payload))]),
+    );
+
+    // Inspect.toInspector payload
+    let call_member = alloc_expr(ast::Expr::Apply(
+        alloc_expr(ast::Expr::Var {
+            module_name: "Inspect",
+            ident: "toInspector",
+        }),
+        &*env.arena.alloc([&*alloc_expr(ast::Expr::Var {
+            module_name: "",
+            ident: payload,
+        })]),
+        roc_module::called_via::CalledVia::Space,
+    ));
+
+    // TODO: change the derived implementation to be something that includes the opaque symbol in
+    // the derivation, e.g. something like
+    //
+    //   \@Opaq payload ->
+    //     Inspect.opaqueWrapper "toString symbol" payload
+
+    // \@Opaq payload -> Inspect.toInspector payload
+    ast::Expr::Closure(
+        env.arena
+            .alloc([Loc::at(DERIVED_REGION, opaque_apply_pattern)]),
+        call_member,
+    )
+}
+
```

Comment:
I'm not really sure when this gets called and if it is implemented correctly. I have worked with a lot of the rest of this code when fixing up inspect, but not this function.

I would assume it is wrong, should it just call Inspect.opaque which will end up outputting `<opaque>`. It looks to claim it has name info, which could be useful.

Reply:
You can test this by adding a uitest/mono test/gen test that tries to call inspect on an opaque type with just `Op := {} implements [Inspect]`. If it works correctly, you'll get the inspected data for the inner type. I think that's the desired behavior here.

Reply:
Given that works, I guess this is implemented correctly.